patricia/patricia.go

```diff
@@ -28,17 +51,33 @@ type Trie struct {
 	item   Item
 
 	children childList
+	option   *Option
 }
 
 // Public API ------------------------------------------------------------------
 
 // Trie constructor.
 func NewTrie() *Trie {
+	o := defaultOption()
+	return NewTrieWithOption(o)
+}
+
+// Trie constructor with option.
+func NewTrieWithOption(o *Option) *Trie {
 	return &Trie{
-		children: newSparseChildList(),
+		children: newSparseChildList(o.MaxChildrenPerSparseNode),
```

Comment:
@tchap Oh, perhaps I misled you. What is important is not configurable setting (though it might be useful), but patricia tree should always return items in sorted manner when walking. I think I should close this pull request and create another one. Sorry for the mess.